src/legacy/core_plugins/data/public/query/query_bar/components/query_bar_input.tsx

```diff
@@ -336,12 +346,75 @@ export class QueryBarInputUI extends Component<Props, State> {
 
     this.onQueryStringChange(newQueryString);
 
+    this.setState({
+      selectionStart: start + (cursorIndex ? cursorIndex : text.length),
+      selectionEnd: start + (cursorIndex ? cursorIndex : text.length),
+    });
+
     if (type === recentSearchType) {
       this.setState({ isSuggestionsVisible: false, index: null });
       this.onSubmit({ query: newQueryString, language: this.props.query.language });
     }
   };
 
+  private handleNestedFieldSyntaxNotification = (suggestion: AutocompleteSuggestion) => {
+    if (
+      'field' in suggestion &&
+      suggestion.field.subType &&
+      suggestion.field.subType.nested &&
+      !this.props.store.get('kibana.KQLNestedQuerySyntaxInfoOptOut')
+    ) {
+      const { notifications } = this.props.kibana.services;
+
+      const onKQLNestedQuerySyntaxInfoOptOut = (toast: Toast) => {
+        if (!this.props.store) return;
+        this.props.store.set('kibana.KQLNestedQuerySyntaxInfoOptOut', true);
+        notifications!.toasts.remove(toast);
+      };
+
+      if (notifications) {
+        const toast = notifications.toasts.add({
+          title: this.props.intl.formatMessage({
+            id: 'data.query.queryBar.KQLNestedQuerySyntaxInfoTitle',
+            defaultMessage: 'KQL nested query syntax',
+          }),
+          text: (
+            <div>
+              <p>
+                <FormattedMessage
```

Comment:
Do we have a standard for translating strings? I though we were moving towards i18n.translate syntax